### PR TITLE
Feat(OperationOrder): #MLYSTOR-597 change display order operation to match order waiting

### PR DESCRIPTION
**Pull request by Rseuret**
## Describe your changes
display order operation to match order waiting

/!\Pr urgente donc les retours non bloquants ne seront peut être pas traités
## Checklist tests
check if each column works properly

## Issue ticket number and link
https://jira.support-ent.fr/browse/MLYSTOR-597

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/lystore/service/impl/DefaultOrderService.java [hunk 1]
```diff
@@ -1663,10 +1663,11 @@ public JsonArray filterValidOrders(JsonArray orders, List<String> filters) {
     @Override
     public void listOrderWaiting(List<String> idCampaigns, List<String> filters, Handler<Either<String, JsonArray>> handler) {
 
-        String query = "SELECT oce.id, oce.price, oce.tax_amount, oce.amount, oce.creation_date, oce.id_campaign, oce.id_structure, oce.name, oce.summary, oce.description," +
```
Comment (Rseuret) [inline src/main/java/fr/openent/lystore/service/impl/DefaultOrderService.java:1666]: indentation

File: src/main/java/fr/openent/lystore/service/impl/DefaultOperationService.java [hunk 2]
```diff
@@ -502,57 +505,132 @@ private JsonObject getDeletionRegion(JsonArray ordersRegionIds) {
...
                 "          orders.\"price TTC\",  " +
-                "          orders.name,  " +
+                "          orders.name," +
+                "          orders.priceht , " +
+                "          orders.tax_amount , " +
                 "          orders.id_structure,  " +
                 "          c.name," +
+                " orders.price_proposal," +
                 " orders.price_proposal ," +
                 " orders.override_region," +
```
Comment (lapeyronnie) [inline src/main/java/fr/openent/lystore/service/impl/DefaultOperationService.java:579]: est-ce normal que price_proposal, soit présent deux fois ?

File: src/main/resources/public/ts/controllers/main.ts [hunk 3]
```diff
@@ -392,6 +392,8 @@ export const mainController = ng.controller('MainController', ['$scope', 'route'
                 }
                 $scope.structures = new Structures();
                 await $scope.structures.sync();
+                await $scope.structures.getStructureType();
+
                 $scope.operation = await $scope.operations.all.find(operationFound => operationFound.id.toString() === params.idOperation.toString());
                 $scope.ordersClientByOperation = await $scope.operation.getOrders($scope.structures.all);
                 template.open('administrator-main', 'administrator/operation/operation-container');
```
Comment (lapeyronnie) [inline src/main/resources/public/ts/controllers/main.ts:399]: ce n'est pas ordersClientByOperation.all ?

**Review by lapeyronnie — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/directives/display-order-list/display-order-list-table/display-order-list-table.html [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            <tbody bottom-scroll="vm.scrollDisplay.limitTo = vm.scrollDisplay.limitTo + 25">
+            <tr ng-repeat="order in vm.displayedOrders.all | orderBy:sort.order.type:sort.order.reverse | limitTo:vm.scrollDisplay.limitTo" post-ng-repeat
+                ng-class="{ orderEditByRegion: order.typeOrder === 'region' }">
+                <td workflow="lystore.manager" class="table-selector">
+                    <label class="checkbox">
+                        <input type="checkbox" ng-model="order.selected">
+                        <span></span>
+                    </label>
+                </td>
+                <td ng-if="vm.tableFields[0].display"> <div class="smallTabElem"> [[order.structure.uai]]</div></td>
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/directives/display-order-list/display-order-list-table/display-order-list-table.html:33]: tu es obligé de gérer ça avec des index ? l'idéal serait un ng-repeat

**Review by qpic-cgi — APPROVED** [review]

File: src/main/resources/public/ts/directives/display-order-list/display-order-list-table/display-order-list-table.html (hunk 4 above)
Comment (Rseuret) [reply src/main/resources/public/ts/directives/display-order-list/display-order-list-table/display-order-list-table.html:33]: oui j'ai vu avec jin et je suis obligé de passé par l index

File: src/main/resources/public/ts/controllers/main.ts (hunk 3 above)
Comment (Rseuret) [reply src/main/resources/public/ts/controllers/main.ts:399]: je ne comprends pas le commentaire

File: src/main/resources/public/ts/directives/display-order-list/display-order-list-table/display-order-list-table.html (hunk 4 above)
Comment (JinDouang) [reply src/main/resources/public/ts/directives/display-order-list/display-order-list-table/display-order-list-table.html:33]: Yes vu avec moi, c'pas gênant pour l'instant

File: src/main/resources/public/ts/directives/display-order-list/display-order-list-table/display-order-list-table.directive.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    tableFields: typeof tableFields;
+}
+
+interface IDirectiveScope extends IScope {
+    vm: IDirectiveProperties;
+}
+
+class Controller implements ng.IController, IViewModel {
+    lang: typeof lang;
+    allOrdersSelected:boolean
```
Comment (JinDouang) [inline src/main/resources/public/ts/directives/display-order-list/display-order-list-table/display-order-list-table.directive.ts:35]: ;

File: src/main/resources/public/ts/directives/display-order-list/display-order-list-table/display-order-list-table.directive.ts [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    scrollDisplay: {
+        limitTo: number
+    }
```
Comment (JinDouang) [inline src/main/resources/public/ts/directives/display-order-list/display-order-list-table/display-order-list-table.directive.ts:38]: ;

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**